core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java

```diff
@@ -29,4 +29,14 @@ public interface SpooledSegmentHandle
     String encoding();
 
     Instant expirationTime();
+
+    default long rows()
+    {
+        return -1;
```

Comment:
Did you read K&R ANSI C recently?

Reply:
It does not seem that metrics should be part of handle. Let's chat about it.

Reply:
why not?

Reply:
Ok, I will leave this information on the `create` span where we have this information in the `SpoolingContext`